packages/core/src/components/DataTable.story.tsx

```diff
@@ -82,6 +93,60 @@ const defaultEditCallback = (
   action('this callback has access to row, key, newVal and event');
 };
 
+export interface SearchDemoProps {
+  data: IndexedParentRow[];
+}
+
+export class SearchDemo extends React.Component {
+  state = {
+    search: '',
+    data: generateRandomData(),
+  };
+
+  handleChange = (value: string) => {
+    this.setState({
+      search: value,
+    });
+  };
+
+  private handleNewData = () => {
+    this.setState({
+      data: generateRandomData(),
+    });
+  };
+
+  filter = (search: string) => (data: IndexedParentRow[]): IndexedParentRow[] => {
+    // @ts-ignore
```

Comment:
what's this for?

Reply:
There's an example use case in the `A table with a search box` story.

Previously I intended for users to write any kind of filter logic in the container component and just update the data prop each time.

In this new model, updating the underlying data implies that an entirely new data set has actually been added, which means we reset selected/expanded state. This would be undesirable if a user is just searching through an input field.

Letting filtering happen inside the data table means we can handle that kind of use case without actually updating the `data` prop, and get to have selected/expanded state persist.

Reply:
aah! that really helps. i know this is just a `story`, but maybe a small comment around this ignore/what this is letting happen would help people looking at the story code? sounds like: let filtering happen without mutating `data` and persist state ?

Reply:
Got it, I'll go ahead and add that, thanks for flagging this!